ega-data-api-res/src/main/java/eu/elixir/ega/ebi/reencryptionmvc/service/internal/LocalEGAArchiveServiceImpl.java

```diff
@@ -69,22 +69,23 @@ public ArchiveSource getArchiveFile(String id, HttpServletResponse response) {
         long size = egaFile.getFileSize();
         String header = egaFile.getHeader();
         try {
-            String privateKey = keyService.getPrivateKey(headerFactory.getKeyIds(header.getBytes()).iterator().next()); // select first subkey
-            Map.Entry<String, String> parsedHeader = parseHeader(header, privateKey);
+            byte[] headerBytes = Hex.decodeHex(header.toCharArray());
+            Collection<String> keyIds = headerFactory.getKeyIds(headerBytes);
+            String privateKey = keyService.getPrivateKey(new BigInteger(Hex.decodeHex(keyIds.iterator().next().toCharArray())).toString()); // select first subkey
```

Comment:
The key id from the my ega.pub is 012C3737B2ED3BA5 extracted with gpg command line.
While the key id returned by reading the code from the java code is 12c3737b2ed3ba5.

It seems that these key ids should be 64 bits so 16 hexadecimal digits.
There seems also to be possible to use 32 bits but they seem insecure: https://evil32.com/

Reply:
So that does not get you the right keyId from the header, it might have something to do with `headerFactory.getKeyIds` ? 
How can i replicate ?

Reply:
Maybe because it does this conversion: `ids.add(Long.toHexString(pgpPublicKeyEncryptedData.getKeyID()));` it suffers from this: https://stackoverflow.com/a/35335975

Reply:
yes, that seems the case. If you want I can send you the keys you can use to replicate

Reply:
@omllobet but that is not related to this issue is it?
Open an issue on https://github.com/uio-bmi/crypt4gh if you think it is necessary or maybe @dtitov can help solve this. The question in the end does this PR solves the issue it was set to solve ?
I do agree that keyID extracted by the library is confusing, so either we fix the library or we chose a new one.

Reply:
@blankdots yes sure, that issue can be solved in another PR in the library. I am approving this PR

Reply:
Did an issue here: https://github.com/uio-bmi/crypt4gh/issues/3

Reply:
That issue is fixed how is it now @omllobet ?

Reply:
Yes sure, I already approved this PR yesterday. I am approving this PR again.